ab3d2_source/c/draw.c

```diff
@@ -0,0 +1,132 @@
+#include "draw.h"
+#include "screen.h"
+#include "system.h"
+
+#include <SDI_compiler.h>
+#include <cybergraphics/cybergraphics.h>
+#include <graphics/gfx.h>
+#include <intuition/intuition.h>
+#include <proto/cybergraphics.h>
+#include <proto/exec.h>
+#include <string.h>  //memset
```

Comment:
My inner optimisation nerd will want to revisit the memory copy and fill routines at some point but this is OK for now.

Reply:
Yeah, this is all non-critical stuff so far. Though the compiler already does a pretty good job at inlining and unrolling memset/memcpy when the size is known at compile time.